doc/source/developing/testing.rst

```diff
@@ -3,503 +3,400 @@
+
+More pytest options can be found by using the ``-h`` option
+
+.. code-block:: bash
+
+    $ pytest -h
 
 .. _answer_testing:
 
 Answer Testing
 --------------
 
-What do Answer Tests Do
+What Do Answer Tests Do
 ^^^^^^^^^^^^^^^^^^^^^^^
 
-Answer tests test **actual data**, and many operations on that data, to make
-sure that answers don't drift over time.  This is how we test frontends, as
-opposed to operations, in yt.
+Answer tests use `actual data <https://yt-project.org/data/>`_ to test reading, writing, and various manipulations of that data.
+
+In order to ensure that each of these operations are performed correctly, yt has a set of yaml files containing the known correct results of having performed these operations. These files are called gold standard answer files. More generally, an answer file is a yaml file containing the results of having run the answer tests.
+
+When the answer tests are run, their output is compared to the gold standard answers to ensure that the results of operating on data do not change over time.
 
 .. _run_answer_testing:
 
 How to Run the Answer Tests
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-The very first step is to make a directory and copy over the data against which
-you want to test. Next, add the config parameter ``test_data_dir`` pointing to
-directory with the test data you want to test with, e.g.:
+In order to run the answer tests locally:
+
+* Create a directory to hold the data required by the answer tests one wishes to run
+
+* Fill this directory with the required data
+
+Next, yt needs to be made aware of where it can find the data. This is done by setting the config parameter ``test_data_dir`` to the
+directory with the test data downloaded from https://yt-project.org/data/. For example,
 
 .. code-block:: bash
 
    $ yt config set yt test_data_dir /Users/tomservo/src/yt-data
 
-We use a number of real-world datasets for the tests that must be downloaded and
-unzipped in the ``test_data_dir`` path you have set. The test datasets, can be
-downloaded from https://yt-project.org/data/. We do not explicitly list the
-datasets we use in the tests here because the list of necessary datasets changes
-regularly, instead you should take a look at the tests you would like to run and
-make sure that the necessary data files are downloaded before running the tests.
+To run the answer tests locally, you must first generate a set of gold standard answer files using a "known good" version of yt. Once done, then update to the version of yt you want to test and
+run the tests again so that the results can be compared to those contained in the gold standard answer files.
 
-To run the answer tests, you must first generate a set of test answers locally
-on a "known good" revision, then update to the revision you want to test, and
-run the tests again using the locally stored answers.
+As an example, let's focus on running the answer tests for the tipsy frontend.
```

Comment:
It would be good to talk about the generation of test answers here, and how one would get newly generated answers from CI if they write new tests.